plugins/digitalocean/resoto_plugin_digitalocean/client.py

```diff
@@ -316,6 +323,75 @@ def list_ssh_keys(self) -> List[Json]:
     def list_tags(self) -> List[Json]:
         return self._fetch("/tags", "tags")
 
+    @retry(
+        stop_max_attempt_number=10,
+        wait_exponential_multiplier=3000,
+        wait_exponential_max=300000,
+        retry_on_exception=retry_on_error,
+    )
+    def get_tag_count(self, tag_name: str) -> Optional[int]:
+        url = f"{self.do_api_endpoint}/tags/{tag_name}"
+        response = requests.get(url, headers=self.headers, allow_redirects=True)
+        if response.status_code == 404:
+            return -1
+        if self.check_status_code(response):
+            return (
+                response.json()
+                .get("tag", {})
+                .get("tag", {})
+                .get("resources", {})
+                .get("count", 0)
+            )
+        return None
+
+    @retry(
+        stop_max_attempt_number=10,
+        wait_exponential_multiplier=3000,
+        wait_exponential_max=300000,
+        retry_on_exception=retry_on_error,
+    )
+    def create_tag(self, tag_name: str) -> bool:
+        url = f"{self.do_api_endpoint}/tags"
+        response = requests.post(url, headers=self.headers, json={"name": tag_name})
+        return self.check_status_code(response)
+
+    @retry(
+        stop_max_attempt_number=10,
+        wait_exponential_multiplier=3000,
+        wait_exponential_max=300000,
+        retry_on_exception=retry_on_error,
+    )
+    def tag_resource(self, tag_name: str, resource_type: str, resource_id: str) -> bool:
```

Comment:
What happens if the tag `tag_name` does not exist?

Reply:
the method will return false. It is a low-level interface that should be used together with add_tag and get_tag_count.